client/admin/content-types/index.jsx

```diff
@@ -7,21 +7,37 @@ import {
   required,
   SelectInput,
   SimpleForm,
-  SimpleFormIterator,
   TextInput,
   FormDataConsumer,
 } from 'react-admin';
 import BookIcon from '@material-ui/icons/Book';
 import { ContentTypeList } from './ContentTypeList';
-import {ImageInputConfig, ImageField} from '../components/ImageInput';
+import { ImageInputConfig } from '../components/ImageInput';
+import { ReferenceInputConfig } from '../components/ReferenceInput';
+
 import get from 'lodash/get';
+import OrderedFormIterator from '../components/OrderedFormIterator';
+
 const ContentTypeTitle = ({ record }) => {
   return <span>Content Type {record ? `"${record.type}"` : ''}</span>;
 };
-const getFieldConfig = (fieldType, source) => {
+const getFieldConfig = (fieldType, source, getSource, scopedFormData) => {
+  // React Admin warning: You're using a FormDataConsumer inside an ArrayInput and you did not called
+  // the getSource function supplied by the FormDataConsumer component.
+  // This is required for your inputs to get the proper source
+  // but we don't need get source every time, sometimes we return empty component
+  // and when we need, we call it in child components
+
+  const _source = typeof getSource === 'function' ? getSource() : null;
```

Comment:
looks like `getSource` is not call only in default case and this line can be moved to `default:`